### PR TITLE
Added metadata-crawler dockerfile. Refactor container workflows

### PR DESCRIPTION
**Pull request by jcortejoso**
### Description

Blockscout metadata crawler is still an in-use tool to fetch celo explorer metadata from validators and validator groups.
The dockerfile for this was deleted (by error) in #10124.

This PR aims to restore the dockerfile, with optimizations/refactoring to reduce the number of packages and dependencies included and also having an updated image with reduced present vulnerabilities.

Apart from that, I've unified different `container-*` workflows in a single `containers.yaml` workflow, trying to improve readability and maintainability.

### Tested

Tested in baklava. It executed correctly (although there were not any metadata change, but I expect to work normally).

### Backwards compatibility

No backwards compatibility issue detected

### CODE REVIEW

**Comment by aaronmgdr** [comment]
hmmi was thinking the meta data crawler was close to deprecated and had included it in the packages moved #10829  to the [new repo for sdks](https://github.com/celo-org/developer-tooling).  (with the thought that it would be removed in the next few months) 

Do you want it to remain in the monorepo? where you could then move it to it own repo if its being kept around.

**Comment by jcortejoso** [comment]
> hmmi was thinking the meta data crawler was close to deprecated and had included it in the packages moved to the new repo for sdks. (with the thought that it would be remove in the next few months)
> 
> Do you want it to remain in the monorepo? where you could then move it to it own repo if its being kept around.

Talked with Valentin and seem we still need it. It is probably only on maintainability mode and we won't be working on new features, so I'd expect very little job on its code (if any).

We can move to its own repo it that will make easier monorepo work.

File: .github/workflows/containers.yaml [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+    uses: celo-org/reusable-workflows/.github/workflows/container-cicd.yaml@v1.13.1
+    needs: changed-files
+    name: Build us-west1-docker.pkg.dev/devopsre/dev-images/blockscoute-metadata-crawler:testing
+    if: |
+      github.ref != 'refs/heads/master' &&
+      contains(needs.changed-files.outputs.all_modified_files, ',dockerfiles/metadata-crawler')
+    with:
+      workload-id-provider: projects/1094498259535/locations/global/workloadIdentityPools/gh-celo-monorepo/providers/github-by-repos
+      service-account: 'celo-monorepo-dev@devopsre.iam.gserviceaccount.com'
+      artifact-registry: us-west1-docker.pkg.dev/devopsre/dev-images/blockscoute-metadata-crawler
```
Comment (alvarof2) [inline .github/workflows/containers.yaml]: `blockscoute` is correct??

File: .github/workflows/containers.yaml [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+    uses: celo-org/reusable-workflows/.github/workflows/container-cicd.yaml@v1.13.1
+    needs: changed-files
+    name: Build us-west1-docker.pkg.dev/devopsre/dev-images/blockscoute-metadata-crawler:latest
+    if: |
+      github.ref == 'refs/heads/master' &&
+      contains(needs.changed-files.outputs.all_modified_files, ',dockerfiles/metadata-crawler')
+    with:
+      workload-id-provider: projects/1094498259535/locations/global/workloadIdentityPools/gh-celo-monorepo/providers/github-by-repos
+      service-account: 'celo-monorepo-dev@devopsre.iam.gserviceaccount.com'
+      artifact-registry: us-west1-docker.pkg.dev/devopsre/dev-images/blockscoute-metadata-crawler
```
Comment (alvarof2) [inline .github/workflows/containers.yaml]: Ditto about `blockscoute`. Also, is the "prod" (`main` branch) image also supposed to go to `dev-images`??

**Review by alvarof2 — APPROVED** [review]

File: .github/workflows/containers.yaml (hunk 2 above)
Comment (jcortejoso) [reply .github/workflows/containers.yaml]: Thanks for the typo.

The registry I thought that for very "minor" images maybe the effort of adding a new registry does not worth it. But maybe is better to follow same patter everywhere.

File: .github/workflows/containers.yaml (hunk 1 above)
Comment (jcortejoso) [reply .github/workflows/containers.yaml]: no :eye:

**Pull request merged**